maplibre/src/style/layer.rs

```diff
@@ -29,6 +30,32 @@ pub struct LinePaint {
     // TODO a lot
 }
 
+pub enum PaintType {
+    Argb(Alpha<EncodedSrgb<f32>>),
+    Raster(RasterLayer),
```

Comment:
A raster layer is not related to the color. The a raster layer should be another case for the `LayerPaint` enum.